packages/web-components/src/components/dotcom-shell/dotcom-shell-container.ts

```diff
@@ -0,0 +1,256 @@
+
+  private _createFooterRenderRoot() {
+    const footer = this.ownerDocument.createElement(`${ddsPrefix}-footer-container`);
+    this.parentNode?.insertBefore(footer, this.nextSibling);
+    return footer;
+  }
+
+  private _createMastheadRenderRoot() {
+    const masthead = this.ownerDocument.createElement(`${ddsPrefix}-masthead-container`);
+    this.parentNode?.insertBefore(masthead, this);
+    return masthead;
+  }
+
+  /**
+   * `true` to activate the search box.
+   */
+  @property({ attribute: 'activate-search' })
+  activateSearch = false;
+
+  /**
+   * The profile items for authenticated state.
+   */
+  @property({ attribute: false })
+  authenticateProfileItems?: MastheadProfileItem[];
+
+  /**
+   * The brand name.
+   */
+  @property({ attribute: 'brand-name' })
+  brandName?: string;
+
+  /**
+   * The g11n collator to use for sorting contry names.
+   */
+  @property({ attribute: false })
+  collatorCountryName = new Intl.Collator();
+
+  /**
+   * The throttle timeout to run query upon user input.
+   */
+  @property({ type: Number })
+  inputTimeout?: number;
+
+  /**
+   * The footer links.
+   */
+  @property({ attribute: false })
+  footerLinks: BasicLinkSet[] = [];
+
+  /**
+   * The language to show in the UI.
+   */
+  @property({ attribute: 'lang-display' })
+  langDisplay?: string;
+
+  /**
+   * The language used for query.
+   */
+  @property()
+  language?: string;
+
+  /**
+   * The legal nav links.
+   */
+  @property({ attribute: false })
+  legalLinks: BasicLink[] = [];
+
+  /**
+   * The locale list.
+   */
+  @property({ attribute: false })
+  localeList?: LocaleModalLocaleList;
+
+  /**
+   * The nonce used for logging in.
+   */
+  @property({ attribute: 'login-nonce' })
+  loginNonce?: string;
+
+  /**
+   * The link target to the masthead logo.
+   */
+  @property({ attribute: 'logo-href' })
+  logoHref?: string;
+
+  /**
+   * The `aria-label` attribute for the top-level container.
+   */
+  @property({ attribute: 'masthead-label' })
+  mastheadLabel?: string;
+
+  /**
+   * The `aria-label` attribute for the menu bar UI.
+   */
+  @property({ attribute: 'menu-bar-label' })
+  menuBarLabel?: string;
+
+  /**
+   * The `aria-label` attribute for the header menu button in its active state.
+   */
+  @property({ attribute: 'menu-button-label-active' })
+  menuButtonLabelActive?: string;
+
+  /**
+   * The `aria-label` attribute for the header menu button in its active state.
+   */
+  @property({ attribute: 'menu-button-label-inactive' })
+  menuButtonLabelInactive?: string;
+
+  /**
+   * `true` to open the locale modal.
+   */
+  @property({ type: Boolean, attribute: 'open-locale-modal' })
+  openLocaleModal = false;
+
+  /**
+   * Footer size.
+   */
+  @property({ reflect: true, attribute: 'footer-size' })
+  footerSize?: FOOTER_SIZE;
+
+  /**
+   * The profile items for unauthenticated state.
+   */
+  @property({ attribute: false })
+  unauthenticatedProfileItems?: MastheadProfileItem[];
```

Comment:
@kennylam offhand do you know if we're getting these from the translation service? I don't think it's split out from the other navigation data, not to say if we should deviate from this or not for the web component implementation of it?

Reply:
@asudoh I believe this comes from the translation data service: https://github.ibm.com/webstandards/ibm-dotcom-translations/blob/8914722a0fb425547d69e14989e4ec5693c2f7ed/translations/masthead-footer/jsononly/usen.json#L1320

We should note that this is used for either SSR or for testing.

Reply:
I see - Updated: https://github.com/carbon-design-system/ibm-dotcom-library/pull/3505/commits/a5f588240d67d327c90e80a21a43e3122d2814ef